cpp/examples/CMakeLists.txt

```diff
@@ -89,3 +89,10 @@ foreach(example
   add_executable(${example} ${example}.cpp)
   target_link_libraries(${example} Proton::cpp Threads::Threads)
 endforeach()
+
+if (ENABLE_OPENTELEMETRYCPP)
+  set(example tracing_demo)
+  add_executable(${example} ${example}.cpp)
+  include_directories(${CMAKE_SOURCE_DIR}/exporters/jaeger/include)
+  target_link_libraries(${example} Proton::cpp opentelemetry_trace opentelemetry_exporter_jaeger_trace)
+endif()
```

Comment:
As far as I can tell the tracing example should compile without any special attention to whether Opentelemetry is compiled into Proton or not. Obviously if there is no real tracing library then this example will do no tracing but I don't think any of the stuff in the conditional should be necessary - if it is necessary then we need to debug why and fix that.(excluding the `add_executable(..)` which I'm suggesting could just go directly into the single threaded examples foreach above.